core/dbt/task/test.py

```diff
@@ -79,37 +92,52 @@ def execute_test(
         table = result['table']
         num_rows = len(table.rows)
         if num_rows != 1:
-            num_cols = len(table.columns)
-            # since we just wrapped our query in `select count(*)`, we are in
-            # big trouble!
             raise InternalException(
                 f"dbt internally failed to execute {test.unique_id}: "
-                f"Returned {num_rows} rows and {num_cols} cols, but expected "
-                f"1 row and 1 column"
+                f"Returned {num_rows} rows, but expected "
+                f"1 row"
+            )
+        num_cols = len(table.columns)
+        if num_cols != 5:
+            raise InternalException(
+                f"dbt internally failed to execute {test.unique_id}: "
+                f"Returned {num_cols} columns, but expected "
+                f"5 columns"
             )
-        return int(table[0][0])
+
+        test_result_data: PrimitiveDict = dict(zip(table.column_names, table.rows[0]))
+
+        return TestResult.from_dict(test_result_data)
 
     def execute(self, test: CompiledTestNode, manifest: Manifest):
-        failed_rows = self.execute_test(test, manifest)
+        result = self.execute_test(test, manifest)
```

Comment:
Do we need to pass through the `error_if` and `warn_if` strings as columns in the result? Or could we access them below as properties of the test node? e.g.
```
message = f'Got {num_errors}, configured to fail if {test.config.error_if}'
message = f'Got {num_errors}, configured to warn if {test.config.warn_if}'
```